### ISSUE TITLE
The user name or passphrase you entered is not correct.

### ISSUE THREAD

**Issue by SilverFoxA** [desktop/desktop#3263]
### Description

Trying to connect to GitHub Desktop on Mac OS High Seria but fails to connect and provides a pop-up screen with the following message "The username or passphrase you entered is not correct".

### Version

**GitHub Desktop version:** 1.0.8

**OS version:** Mac OS 10.13

### Steps to Reproduce

1. Download GitHub desktop from the official website
1. Click on Clone repo or preferences to login to GitHub.com
1. Enter credentials and login

**Expected behavior:** Login to Github.com and fetch the repositories assigned to the user's account

**Actual behavior:** Login process is not complete. Give error "The user name or passphrase you entered is not correct."

**Reproduces how often:** Always (100%)

### Logs

2017-11-08T06:07:19.570Z - info: [ui] [Dispatcher] requesting authenticated user
2017-11-08T06:07:21.592Z - info: [ui] [SignInStore] account resolved
2017-11-08T06:07:21.592Z - info: [ui] [AppStore] adding account SilverFoxA (Abhijit Das) to store
2017-11-08T06:07:21.631Z - error: [ui] Error adding account 'SilverFoxA'
Error: The user name or passphrase you entered is not correct.

**Comment by shiftkey** [comment]
@SilverFoxA a couple of questions to help reproduce the issue:

> Enter credentials and login

Can you share a screenshot of the dialog you used to enter your credentials? I want to confirm it's the GitHub flow and not the generic account flow.

Also, does your account have 2FA enabled?

**Comment by shiftkey** [comment]
@SilverFoxA actually, it looks like this error is really helpful:

```
2017-11-08T06:07:21.631Z - error: [ui] Error adding account 'SilverFoxA'
Error: The user name or passphrase you entered is not correct.
```

The source of this error is this function, which is interoperating with the keychain:

https://github.com/desktop/desktop/blob/7412eda2bb171f1eabd3a25f4c5a959e9592eaea/app/src/lib/stores/accounts-store.ts#L96-L106

Could something in High Sierra be restricting access to the keychain?

**Comment by w1w1w2w2** [comment]
[3331111111111111111111111111.txt](https://github.com/desktop/desktop/files/1460999/3331111111111111111111111111.txt)

**Comment by yassermog** [comment]
I have the same problem

**Comment by BaldwinLi** [comment]
I have same issue after upgraded to macOS Hight Sierra 10.13.1
So far I have not ability to push any code into my repositories, seems very terrible.

**Comment by nathankonrad1** [comment]
Hey guys, any update on this ?

we are totally in bad situation: cant login (even singing using browser), commit, or any other action.

Can you guys please help ?

**Comment by joshaber** [comment]
>cant login (even singing using browser)

That would seem to indicate a problem with the user account, not GitHub Desktop. I'd recommend you contact http://github.com/support.

@yassermog @BaldwinLi Could you provide your logs please?

**Comment by basheerMalaaAccolade** [comment]
I have the same problem

**Comment by joshaber** [comment]
Whoops, looks like our bot got a little too aggressive. Re-opening.

**Comment by blong1996** [comment]
Hey, having this same issue. Unable to push to any Github.com repos due to this authentication bug paired with High Sierra. Any idea on a temporary fix?

**Comment by joshaber** [comment]
@blong1996 Could you provide your logs? They should be at `~/Library/Application Support/GitHub Desktop/logs/*.desktop.production.log`

**Comment by AtomicCons** [comment]
Running 10.13 no issues. upgrading to 10.13.1 right now. Ill post pack my results.

**Comment by blong1996** [comment]
[2017-11-15.desktop.production.log](https://github.com/desktop/desktop/files/1473309/2017-11-15.desktop.production.log)
@joshaber

**Comment by blong1996** [comment]
Just to confirm. High Sierra: 10.13.1 & GitHub Desktop 1.0.8

**Comment by AtomicCons** [comment]
High Sierra is not the issue. Running 10.13.1 with GHD 1.0.8. No issues noted

**Comment by blong1996** [comment]
🤷🏽‍♂️ @AtomicCons

**Comment by ofkindness** [comment]
I have same issue with 1.0.8 (Mac OS 10.13.1) 

```2017-11-15T08:46:18.635Z - info: [ui] [AppStore] loading 0 repositories from store
2017-11-15T08:46:18.826Z - info: [ui] [Welcome] no sign in step found. ignoring...
2017-11-15T08:46:18.841Z - info: [ui] [Welcome] no sign in step found. ignoring...
2017-11-15T08:46:19.046Z - info: [ui] [Welcome] no sign in step found. ignoring...
2017-11-15T08:46:19.400Z - info: [ui] [Welcome] no sign in step found. ignoring...
2017-11-15T08:46:19.500Z - info: [ui] launching: 1.0.8 (Mac OS 10.13.1)
2017-11-15T08:46:19.555Z - info: [ui] [Welcome] no sign in step found. ignoring...
2017-11-15T08:46:25.209Z - info: [ui] [Welcome] advancing to step: SignInToDotCom
2017-11-15T08:46:25.217Z - info: [ui] [Welcome] current sign in state not found. ignoring...
2017-11-15T08:46:36.200Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2017-11-15T08:46:54.472Z - info: [ui] [Welcome] kind TwoFactorAuthentication is the same as TwoFactorAuthentication. ignoring...
2017-11-15T08:46:55.236Z - info: [ui] [AppStore] adding account ofkindness (null) to store
2017-11-15T08:46:55.254Z - info: [ui] [Welcome] advancing to step: ConfigureGit
2017-11-15T08:46:55.351Z - error: [ui] Error adding account 'ofkindness'
Error: The user name or passphrase you entered is not correct.
2017-11-15T08:46:55.357Z - info: [ui] [Welcome] no sign in step found. ignoring...
2017-11-15T08:47:14.360Z - info: [ui] [Welcome] advancing to step: UsageOptOut
2017-11-15T08:47:18.976Z - info: [ui] Opt out reported.
2017-11-15T08:48:40.375Z - info: [ui] [AppStore] adding account ofkindness (null) to store
2017-11-15T08:48:40.418Z - error: [ui] Error adding account 'ofkindness'
Error: The user name or passphrase you entered is not correct.
2017-11-15T08:53:05.820Z - info: [ui] [AppStore] adding account ofkindness (null) to store
2017-11-15T08:53:05.886Z - error: [ui] Error adding account 'ofkindness'
Error: The user name or passphrase you entered is not correct.```

**Comment by joshaber** [comment]
It sure looks like something's screwy with the Keychain.

Try opening Keychain Access.app. Right-click on the `login` keychain and try locking it and then unlocking it again.

**Comment by ofkindness** [comment]
@joshaber That's the best trick I've ever heard of in the history

**Comment by AtomicCons** [comment]
LOL. @ofkindness,
Is the latest command line tool for OS X installed?

**Comment by joshaber** [comment]
@ofkindness 😆 did that work?

**Comment by blong1996** [comment]
@joshaber it worked ! Thanks man

**Comment by shiftkey** [comment]
Seeing how I've not heard any new reports for a few days, I'm going to close this out and point to @joshaber's lock/unlock trick here https://github.com/desktop/desktop/issues/3263#issuecomment-344590213 or @AtomicCons's update to the command line tools here https://github.com/desktop/desktop/issues/3263#issuecomment-344605189

**Comment by nickmilani** [comment]
Honestly.. the locking and unlocked actually worked. I have no idea what I just did, but HOORAY

**Comment by mtrellet** [comment]
As extra information, I'm running OSX El Capitan (10.11.6) and GitHub Desktop 1.0.9 and bumped into the very same issue this morning. Login directly in Desktop or via the web browser was resulting in:
`Error: The user name or passphrase you entered is not correct.`

Logs:

```
2017-11-20T09:47:05.124Z - error: [ui] Error getting token for 'GitHub - https://api.github.com'. Skipping.
Error: The user name or passphrase you entered is not correct.
2017-11-20T09:47:05.234Z - info: [ui] [AppStore] loading 11 repositories from store
2017-11-20T09:47:05.830Z - info: [ui] launching: 1.0.9 (Mac OS 10.11.6)
2017-11-20T09:47:20.827Z - info: [ui] [AppStore] adding account mtrellet (Mikael Trellet) to store
2017-11-20T09:47:20.862Z - error: [ui] Error adding account 'mtrellet'
Error: The user name or passphrase you entered is not correct.
2017-11-20T09:48:03.221Z - info: [ui] [AppStore] adding account mtrellet (Mikael Trellet) to store
2017-11-20T09:48:03.247Z - error: [ui] Error adding account 'mtrellet'
Error: The user name or passphrase you entered is not correct.
2017-11-20T09:48:09.679Z - info: [ui] [SignInStore] initializing OAuth flow
2017-11-20T09:48:09.755Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=******
2017-11-20T09:48:23.355Z - info: [ui] [Dispatcher] requesting authenticated user
2017-11-20T09:48:24.520Z - info: [ui] [SignInStore] account resolved
2017-11-20T09:48:24.520Z - info: [ui] [AppStore] adding account mtrellet (Mikael Trellet) to store
2017-11-20T09:48:24.549Z - error: [ui] Error adding account 'mtrellet'
Error: The user name or passphrase you entered is not correct.
```

The locking/unlocking worked perfectly to solve the issue. 👍

**Comment by yaominguo** [comment]
@joshaber  Thank u ! it's worked!

**Comment by andrey-zelenin** [comment]
@joshaber thanks, it helps me too

**Comment by SilverFoxA** [comment]
@shiftkey Sorry for being inactive. Seems like the error occurred because of keychain conflicts.
Solved it by deleting the stored Git creds. from `Keychain` & did a system restart.

**Comment by goldrush-bob** [comment]
I was fighting this login issue this morning for over 2 hours. I locked, then unlocked my keychain and 'poof' fixed the issue like magic! Thanks!!!

**Comment by shatabdichoudhury** [comment]
I ran into the problem and it worked after locking and unlocking the keychain. thank you so much.

**Comment by Jinny-Wang** [comment]
@joshaber  It works!! Thanks a lot.  Any idea for the reasons behind this amazing magic fixing?

**Comment by joshaber** [comment]
Not really 😒 My best guess is that the keychain thinks it is locked but it's really not. So locking and unlocking resets whatever state is confused.

**Comment by goldrush-bob** [comment]
I'd like to add that my experience included issues with my Mail.app, meaning my accounts were suddenly locked out and asking for passwords. The first day, I updated all passwords for my email accounts only to be prompted for passwords again the next day. At that point further research lead me to this thread. So, although I began to suspect the keychain was the culprit, the simple "lock/unlock" hadn't occurred to me. So, once again... thanks for this post and the easy fix!

**Comment by rckrueger** [comment]
josh thanks a million - that lock/unlock trick was the best!!!

**Comment by ethancwb** [comment]
OMG I can't believe so many people is having this issue. It worked for me too! I guess the reason could be when you denied the Keychain.app to remember one of your keys, it actually locks the whole thing in the backend. I denied once today and had this issue.

**Comment by kcharcast** [comment]
@joshaber you handsome devil, that works!

**Comment by Fraer** [comment]
@joshaber Thank you so much !!

**Comment by marciotedesco** [comment]
@joshaber yep that worked here too, High Sierra 10.13.3 - GHD 1.0.13

**Comment by Jamonek** [comment]
Worked for me also!

**Comment by kmetto** [comment]
@joshaber best fix ever 👍🏻

**Comment by darthrevan** [comment]
Thanks man, helped me solve completely unrelated issue 🤣 🤣 🤣 🤣 🤣 🤣

**Comment by mohit-wal** [comment]
Thanks a lot. But I spent more than 5 hours trying to find a solution. Installed fresh version, etc.

**Comment by PRCrawshaw** [comment]
Yep - locking and unlocking keychain did it for me to! - Thanks

**Comment by jcreager** [comment]
@joshaber Thank you!  Unlocking and locking totally worked.

**Comment by infa-amohandas** [comment]
@joshaber It was a good trick. I was going to uninstall and reinstall Github Desktop. Thank you :)

**Comment by sansanbgl** [comment]
@joshaber  thanks, the trick works

**Comment by meetagulati** [comment]
WORKED !!! Woohoo. Thank you !

**Comment by NAZMULBABLU** [comment]
Thanks too

**Comment by bandzejewicz** [comment]
I'm having an issue now as well. I'm signed into github.com. When I try to clone a repository from a URL, I'm getting an "Authentication Failed" message. When I enter my username and password, I get the same prompt.

**Comment by mlapeyre3** [comment]
> It sure looks like something's screwy with Keychain.
> 
> Try opening Keychain Access.app. Right-click on the `login` keychain and try locking it and then unlocking it again.

Pro-tip !!

**Comment by ericgopak** [comment]
> 
> 
> It sure looks like something's screwy with Keychain.
> 
> Try opening Keychain Access.app. Right-click on the `login` keychain and try locking it and then unlocking it again.

After locking the keychain I couldn't unlock it :man_facepalming: 
But after restarting the Mac it got unlocked automatically.

Thanks, it worked!

**Comment by lcomet** [comment]
> It sure looks like something's screwy with Keychain.
> 
> Try opening Keychain Access.app. Right-click on the `login` keychain and try locking it and then unlocking it again.

Awesome! thanks

**Comment by MasterXKDD** [comment]
Dark Messiah

**Comment by xieliying** [comment]
Worked for me, thanks a lot!!!

**Comment by xieliying** [comment]
@joshaber Worked for me, thank you very much!!!

**Comment by hotcoder** [comment]
@joshaber  , tip did not worked for me , any one have still have the issue and got solution

**Comment by ezradt** [comment]
thank you! just what I needed.
Probably the change my computer password made this happen

**Comment by knyto2** [comment]
worked for me too!

**Comment by chrisbalchin** [comment]
+1

**Comment by robindowntivo** [comment]
I am able to login with web browser to githuub. I verified my username and password, but when I try to fetch my repo from Terminal app on OS Mojave I keep getting 'username or password is incorrect' error. I've tried locking/unlocking my keychain, no luck. I have 2 factor authentication enabled, when I login to web I get prompted for a pin sent to my phone, but i see no such prompt from command line login. Is this part of the problem??

**Comment by robindowntivo** [comment]
I found the answer for my issue here: https://github.com/Microsoft/vscode/issues/46964

**Comment by nikunjgajera** [comment]
@joshaber worked for me

**Comment by Abrarhaj** [comment]
ok

**Comment by TimeLimitless666** [comment]
Y988036862@gmail.com

**Comment by 1atAlcone** [comment]
> ### Description
> Trying to connect to GitHub Desktop on Mac OS High Sierra but fails to connect and provides a pop-up screen with the following message "The username or passphrase you entered is not correct".
> 
> ### Version
> **GitHub Desktop version:** 1.0.8
> 
> **OS version:** Mac OS 10.13
> 
> ### Steps to Reproduce
> 1. Download GitHub desktop from the official website
> 2. Click on Clone repo or preferences to login to GitHub.com
> 3. Enter credentials and login
> 
> **Expected behavior:** Login to Github.com and fetch the repositories assigned to the user's account
> 
> **Actual behavior:** Login process is not complete. Give error "The user name or passphrase you entered is not correct."
> 
> **Reproduces how often:** Always (100%)
> 
> ### Logs
> 2017-11-08T06:07:19.570Z - info: [ui] [Dispatcher] requesting authenticated user 2017-11-08T06:07:21.592Z - info: [ui] [SignInStore] account resolved 2017-11-08T06:07:21.592Z - info: [ui] [AppStore] adding account SilverFoxA (Abhijit Das) to store 2017-11-08T06:07:21.631Z - error: [ui] Error adding account 'SilverFoxA' Error: The user name or passphrase you entered is not correct.
> 
> #### Additional Information
